pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -2306,6 +2306,310 @@ def init_arc_param(arctype):
         doc="Operating capacity of disposal site [%]",
     )
 
+    model.p_eta_TruckingEmissionsCoefficient = Param(
+        model.s_A,
+        default=0,
+        initialize={
+            y: model.df_parameters["AirEmissionCoefficients"][(x, y)]
+            for (x, y) in model.df_parameters["AirEmissionCoefficients"]
+            if "Trucking" in x
+        },
+        units=model.model_units[
+            "mass"
+        ],  # Note: Hours are not modeled as a separate unit (see p_pi_Trucking)
+        mutable=True,  # Mutable Param - can be changed in sensitivity analysis without rebuilding the entire model
```

Comment:
I agree that the comment `# Mutable Param - can be changed in sensitivity analysis without rebuilding the entire model` is not needed and is potentially misleading for Params with units, since those are required to be mutable. I will remove all instances of this comment.

As for the discrepancy between the units and the documentation, that is intentional. See the parameter definitions and comments that I have copied which provide more explanation.

    # It is expected that the units for p_tau_trucking are hours, which usually differs from the time units used
    # for flow rates, e.g., day, week, month. Therefore, for the sake of simplicity, no units are defined for p_tau_trucking,
    # as the hr units will cancel out with the units in model.p_pi_Trucking which is the hourly cost of trucking.
    model.p_tau_Trucking = Param(
        model.s_L,
        model.s_L,
        default=12,
        initialize=model.df_parameters["TruckingTime"],
        mutable=True,
        doc="Drive time between locations [hr]",
    )

    ...

    # COMMENT: For this parameter only currency units are defined, as the hourly rate is canceled out with the
    # trucking time units from parameter p_Tau_Trucking. This is to avoid adding an extra unit for time which may
    # be confusing
    model.p_pi_Trucking = Param(
        model.s_L,
        default=(
            max(TruckingHourlyCost_convert_to_model.values()) * 100
            if TruckingHourlyCost_convert_to_model
            else pyunits.convert_value(
                15000,
                from_units=pyunits.USD,
                to_units=model.model_units["currency"],
            )
        ),
        initialize=TruckingHourlyCost_convert_to_model,
        units=model.model_units["currency"],
        doc="Trucking hourly cost (by source) [currency/hr]",
    )

You can see here how we have neglected to explicitly implement the hour unit for trucking times. We just

Reply:
This is now addressed - resolving.